### PR TITLE
Turned on client side tests. Fixed broken tests

### PR DESCRIPTION
**Pull request by TheHollidayInn**
Fixes put_issue_url_here
### Changes

Add client tests to npm run
Fixed broken client tests. 

---

UUID:

### CODE REVIEW

File: website/public/js/services/taskServices.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 
 (function(){
-  var TASK_KEYS_TO_REMOVE = ['_id', 'completed', 'date', 'dateCompleted', 'dateCreated', 'history', 'id', 'streak'];
+  var TASK_KEYS_TO_REMOVE = ['_id', 'completed', 'date', 'dateCompleted', 'dateCreated', 'history', 'id', 'streak', 'createdAt'];
```
Comment (paglias) [inline website/public/js/services/taskServices.js]: dateCreated is not stored anymore in v3, it now uses createdAt

**Comment by crookedneighbor** [comment]
Why do you need setTimeout in the tests? Does the save call not provide a callback or a promise to show when it's done? If it doesn't, why not?

File: website/public/js/controllers/partyCtrl.js [hunk 2]
```diff
@@ -46,7 +46,9 @@ habitrpg.controller("PartyCtrl", ['$rootScope','$scope','Groups','Chat','User','
         }
       }
 
-      Chat.markChatSeen($scope.group._id);
+      if ($scope.group) {
```
Comment (crookedneighbor) [inline website/public/js/controllers/partyCtrl.js]: About time :)

**Comment by TheHollidayInn** [comment]
Well, this is the thing I have been confused about.

For example: The createChallenge function is a promise. But, in the controller tests, I don't just need to test after the callback is returned, I need to test after the callback returns then runs some actions - like User.sync. 

So, these seem to be examples of background calls, so I need to wait for the background calls to finish.

File: website/public/js/controllers/partyCtrl.js (hunk 2 above)
Comment (TheHollidayInn) [inline website/public/js/controllers/partyCtrl.js]: lol, it seemed odd to me as well.

File: website/public/js/services/taskServices.js (hunk 1 above)
Comment (TheHollidayInn) [inline website/public/js/services/taskServices.js]: Okay, I changed createAt in the tests. I wasn't sure if I should remove it from this array for any v2 support.

**Comment by TheHollidayInn** [comment]
Updated!

File: website/public/js/services/taskServices.js (hunk 1 above)
Comment (paglias) [inline website/public/js/services/taskServices.js]: Nah, you can remove it

**Comment by TheHollidayInn** [comment]
Updated!

**Pull request merged**